### DISCUSSION TITLE
Hard to follow conversations in PR review with many iterations

### DISCUSSION THREAD

**Discussion by mrkam2** [community/community#14886] [Pull Requests]
I filed this as a [ticket](https://support.github.com/ticket/1577711) initially, but was asked to start a thread here.

Title: Show a timeline of comments

When there is a long review process on a PR, with rebases and a repeating reviews, it becomes hard to follow.

I have a number of issues and suggestions about them:

- When there is an email notification about a comment, the link in the email doesn't always work. This is probably related to one of the two things:
  - either the UI isn't showing the discussion, replacing it with message like "150 hidden items", and the link from the email isn't working
  - there was a rebase that caused the code comment to be linked to a commit that's no longer in the git.
- Sometimes, when I'm trying to find the comment, I can see the individual comment, but I can't see the original discussion it was linked to (i. e. previous comment), nor I have the "add comment" text box to respond.
- I often want to review my recent comments, and when we have a long discussion, I need to expand "150 hidden items" a number of times before I can find them. The message itself doesn't stand out, so it is hard to notice it when you're scrolling the review.

A possible improvement to this could be adding a timeline of comments, where they're showed in a chronological order, with easy way to see the recent ones, and where every comment is shown in a context of a previous and follow-up discussion and a code fragment it relates to. This is superior to the case where replies to discussions only appear next to the comments that originate the discussion and not in a chronological order. Consider this example:

User A does a review with 30 comments.
User B responds to some of the comments, updates the code, rebases it on the updated main branch and requests another review.
Repeat this a few times.
The task of finding all comments that were added between the last time user A did a review and the current moment is very hard to solve because responses to the comments are shown within previous reviews, not in between.

**Comment by StingyJack** [comment]
I don't even think many iterations are needed to see this problem. Conversations are hard to follow and sometimes just go missing after the first commit following a review.

**Comment by IvanRibakov** [comment]
Here's a related discussion I opened some time ago: https://github.com/github/feedback/discussions/10367

I wonder if it's possible to merge the discussions and combine the votes...
